app/src/pages/Rankings.tsx

```diff
@@ -0,0 +1,253 @@
+import { useTranslation } from "react-i18next";
+import {
+  Box,
+  Container,
+  Theme,
+  ToggleButton,
+  ToggleButtonGroup,
+  Typography,
+  useMediaQuery,
+} from "@mui/material";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+import Stack from "@mui/material/Stack";
+import Switch from "@mui/material/Switch";
+import CircularProgress from "@mui/material/CircularProgress";
+
+import { eventID, Province } from "../components/Types";
+import { getProvinces } from "../components/provinces";
+import { API_BASE_URL, PRODUCTION } from "../components/api";
+import httpClient from "../httpClient";
+import { RankList } from "../components/RankList";
+import { MyCubingIcon } from "../components/MyCubingIcon";
+import useResponsiveQuery from "../components/useResponsiveQuery";
+
+const provinces: Province[] = getProvinces();
+const events: eventID[] = [
+  "333",
+  "222",
+  "444",
+  "555",
+  "666",
+  "777",
+  "333bf",
+  "333fm",
+  "333oh",
+  "clock",
+  "minx",
+  "pyram",
+  "skewb",
+  "sq1",
+  "444bf",
+  "555bf",
+  "333mbf",
+];
+
+export const Rankings = () => {
+  const { t } = useTranslation();
+  const isSmall = useResponsiveQuery("sm");
+
+  const [province, setProvince] = useState<Province | null>(provinces[0]);
+  const [eventId, setEventId] = useState<eventID>("333");
+  const [usingAverage, setUsingAverage] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const [ranking, setRanking] = useState<any | null>(null);
+
+  const switchHandler = (event: {
+    target: { checked: boolean | ((prevState: boolean) => boolean) };
+  }) => {
+    setUsingAverage(event.target.checked);
+  };
+
+  const handleProvinceChange = (
+    event: any,
+    newValue: React.SetStateAction<Province | null>,
+  ) => {
+    setProvince(newValue);
+
+    if (province != null) {
+      console.log(ranking);
+      if (province.id === "qc") {
+        console.log("Vive le Québec libre!");
+      }
+    }
+  };
+
+  const handleEventChange = (
+    event: React.MouseEvent<HTMLElement>,
+    newEvent: eventID | null,
+  ) => {
+    if (newEvent === null) {
+      return;
+    }
+    setEventId(newEvent);
+  };
+  const handleEventChangeMobile = (
+    event: any,
+    newValue: React.SetStateAction<eventID | null>,
+  ) => {
+    if (newValue === null) {
+      return;
+    }
+    setEventId(newValue as eventID);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    setRanking(null);
+    const use_average_str = usingAverage ? "1" : "0";
+
+    (async () => {
+      try {
+        if (
+          eventId === null ||
+          province?.id === null ||
+          usingAverage === null ||
```

Comment:
Isn't `usingAverage` always a boolean?
Can the conditional be simplified to `if (!eventId || !province?.id) ...`?